prototypes/skill-manifest-schema-1-0-0.json

```diff
@@ -134,38 +133,58 @@
         "msAppId": {
           "type": "string",
           "title": "Microsoft App Id",
-          "pattern": "[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{12}$",
+          "pattern": "^[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{12}$",
           "description": "The Microsoft AppId for the skill. This app ID is used to authenticate requests"
         }
       }
     },
-    "eventActivity": {
+    "activityBase": {
       "type": "object",
-      "description": "An activity with Type=Event where the Name property indicates the task that the skill will execute",
+      "description": "An activity with a type",
+      "discriminator": {
+        "propertyName": "type",
+        "mapping": {
+          "event": "#/definitions/eventActivity",
+          "invoke": "#/definitions/invokeActivity",
+          "message": "#/definitions/messageActivity"
+        }
+      },
```

Comment:
oh that's cool shit.  Does VS Code honor this?

Reply:
Thanks Tom! "discriminator" is a openapi/swagger-ism that is not part of json schema (though, NJsonSchema.dll & NSwag happen to allow openapi-ism's, which is what I used for codegen). 

So, vscode is indeed a problem, and we would need to provide additional $schema to understand discriminator.

eg: along the lines of -
https://github.com/OAI/OpenAPI-Specification/blob/master/schemas/v3.0/schema.json
.... I think, just the "discriminator" part.

As json schema is still being standardized, having support in microsoft.bot.schema, or a nswag-equivalent command line tool to generate code from a manifest would be helpful here.

CC: @sbacha @gabog